maud_macros/src/parse.rs

```diff
@@ -178,29 +176,25 @@ impl Parser {
             // Block
             TokenTree::Group(ref group) if group.delimiter() == Delimiter::Brace => {
                 self.advance();
-                ast::Markup::Block(self.block(group.stream(), group.span())?)
+                ast::Markup::Block(self.block(group.stream(), group.span()))
             },
             // ???
             token => {
-                token.span().error("invalid syntax").emit();
-                return Err(());
+                abort!(token, "invalid syntax");
             },
         };
-        Ok(markup)
+        markup
     }
 
     /// Parses and renders a literal string.
-    fn literal(&mut self, lit: &Literal) -> ParseResult<ast::Markup> {
+    fn literal(&mut self, lit: &Literal) -> ast::Markup {
         let content = parse_str::<LitStr>(&lit.to_string())
             .map(|l| l.value())
-            .unwrap_or_else(|_| {
-                lit.span().error("expected string").emit();
-                String::new()  // Insert a dummy value
-            });
-        Ok(ast::Markup::Literal {
+            .unwrap_or_else(|_| abort!(lit, "expected string"));
```

Comment:
But then you don't have a literal to return if the error has triggered. I could return a dummy literal (like `"dummy"`) though; should I?

Reply:
Frankly, I on't see much point in this case. Missed literal should probably be a hard error, otherwise you're risking to spawn god knows how many non-existent errors while trying to recover.

Reply:
The intention is to handle cases like

```rust
input type="number" min=0 max=100;
```

where the only problem is that the `0` and `100` need to be wrapped in quotes.

> A missed literal should probably be a hard error, otherwise you're risking to spawn a cascade of god knows how many non-existent errors while trying to recover.

That is true. But in this context, it's not a *missed* literal, just one of the wrong type. That said, I think the message can be improved... #208

> But then you don't have a literal to return if the error has triggered.

We can return `String::new()` like the original code.

Reply:
> it's not a missed literal, just one of the wrong type

Right. Don't know why I thought it was about missing literals, let's just say that the voices in my head can be wrong sometimes.